modules/c++/math/source/Utilities.cpp

```diff
@@ -0,0 +1,47 @@
+/* =========================================================================
+ * This file is part of math-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2017, MDA Information Systems LLC
+ *
+ * math-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <import/sys.h>
+#include <math/Utilities.h>
+
+namespace math
+{
+size_t nChooseK(size_t n, size_t k)
+{
+    if (n < k)
+    {
+        throw except::Exception(Ctxt("n Choose k undefined for n < k"));
```

Comment:
Add the values of `n` and `k` to the exception message.  That way if this somehow gets triggered 35 levels down of some complex chain and we're seeing this in a log message, we know what values were mistakenly sent in.